bastionzero/service/environments/environments.go

```diff
@@ -38,6 +38,7 @@ type CreateEnvironmentResponse struct {
 
 // ModifyEnvironmentRequest is used to modify an environment
 type ModifyEnvironmentRequest struct {
+	Name        *string `json:"name,omitempty"`
```

Comment:
This API change is technically related to the Terraform provider. Once we merge this PR, we can also update the Terraform provider to permit the `bastionzero_environment` resource's name to change without requiring a replace (current behavior).

We can either add this enhancement to the TF provider now, or make a JIRA ticket and add it later. It's a pretty small change to make (with only 1 required integration test to test it), but I'm fine with coming back to it later if there are other higher priority things.

The changes would look something like this:

1) Remove the `RequiresReplace` plan modifier to permit name to be modified without having to destroy and recreate the environment: https://github.com/bastionzero/terraform-provider-bastionzero/blob/ef3edad008c362b536eb370f4130a2ec4675224c/bastionzero/environment/environment.go#L118-L120
2) Add support for modifying `Name` in the `Update` method: https://github.com/bastionzero/terraform-provider-bastionzero/blob/ef3edad008c362b536eb370f4130a2ec4675224c/bastionzero/environment/resource_environment.go#L155-L163
3) Create a new integration test similar to this one here to test updating an environment's name via the TF provider: https://github.com/bastionzero/terraform-provider-bastionzero/blob/ef3edad008c362b536eb370f4130a2ec4675224c/bastionzero/policy/sessionrecording/resource_sessionrecordingpolicy_test.go#L93-L136

Reply:
Pull request opened here -> https://github.com/bastionzero/terraform-provider-bastionzero/pull/35